<a href="https://colab.research.google.com/github/yash966/Amazon-Sales/blob/main/AmazonSales.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##**Objectives**
* The primary objective of analyzing the Amazon Sales Dataset is delve into product categories, prices, ratings, and sales patterns to identify characteristics that resonate with consumers and propel them to purchase.¶

* Delve into product categories, prices, ratings, and sales patterns to identify characteristics that resonate with consumers and propel them to purchase.

* Translate insights into actionable recommendations that optimize product development, inform marketing strategies, and boost your competitive edge.

* Equip businesses with the knowledge to create products that cater to evolving consumer needs and desires.

* Craft communication strategies that resonate with specific demographics and maximize engagement.

Facilitate a marketplace where products find their perfect match in the hearts of consumers.






## **Importing The Dataset from Kaggle**

In [ ]:
import kagglehub
path = kagglehub.dataset_download("karkavelrajaj/amazon-sales-dataset")

Using Colab cache for faster access to the 'amazon-sales-dataset' dataset.


In [ ]:
path

'/kaggle/input/amazon-sales-dataset'

In [ ]:
# 1. First, check what files were downloaded
import os
print("Files in the dataset:")
print(os.listdir(path))

Files in the dataset:
['amazon.csv']


In [ ]:
# 2. Load the data into a pandas DataFrame
import pandas as pd

# Assuming it's a CSV file (common for Kaggle datasets)
# Replace 'filename.csv' with the actual filename you see
df = pd.read_csv(f"{path}/amazon.csv")
copy = pd.read_csv(f"{path}/amazon.csv")

# If you're not sure of the filename, you can also try:
# df = pd.read_csv(f"{path}/{os.listdir(path)[0]}")

In [ ]:
df.head(5)

product_id                                       product_name  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3  B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4  B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   

                                            category discounted_price  \
0  Computers&Accessories|Accessories&Peripherals|...             ₹399   
1  Computers&Accessories|Accessories&Peripherals|...             ₹199   
2  Computers&Accessories|Accessories&Peripherals|...             ₹199   
3  Computers&Accessories|Accessories&Peripherals|...             ₹329   
4  Computers&Accessories|Accessories&Peripherals|...             ₹154   

  actual_price discount_percentage rating rating_count  \
0       ₹1,099                 64%    4.2       24,269   
1         ₹349                 43%    4.0       43,994   
2       ₹1,899                 90%    3.9        7,928   
3         ₹699                 53%    4.2       94,363   
4         ₹399                 61%    4.2       16,905   

                                       about_product  \
0  High Compatibility : Compatible With iPhone 12...   
1  Compatible with all Type C enabled devices, be...   
2  【 Fast Charger& Data Sync】-With built-in safet...   
3  The boAt Deuce USB 300 2 in 1 cable is compati...   
4  [CHARGE & SYNC FUNCTION]- This cable comes wit...   

                                             user_id  \
0  AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1  AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2  AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3  AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4  AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   

                                           user_name  \
0  Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...   
1  ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...   
2  Kunal,Himanshu,viswanath,sai niharka,saqib mal...   
3  Omkar dhale,JD,HEMALATHA,Ajwadh a.,amar singh ...   
4  rahuls6099,Swasat Borah,Ajay Wadke,Pranali,RVK...   

                                           review_id  \
0  R3HXWT0LRP0NMF,R2AJM3LFTLZHFO,R6AQJGUP6P86,R1K...   
1  RGIQEG07R9HS2,R1SMWZQ86XIN8U,R2J3Y1WL29GWDE,RY...   
2  R3J3EQQ9TZI5ZJ,R3E7WBGK7ID0KV,RWU79XKQ6I1QF,R2...   
3  R3EEUZKKK9J36I,R3HJVYCLYOY554,REDECAZ7AMPQC,R1...   
4  R1BP4L2HH9TFUP,R16PVJEXKV6QZS,R2UPDB81N66T4P,R...   

                                        review_title  \
0  Satisfied,Charging is really fast,Value for mo...   
1  A Good Braided Cable for Your Type C Device,Go...   
2  Good speed for earlier versions,Good Product,W...   
3  Good product,Good one,Nice,Really nice product...   
4  As good as original,Decent,Good one for second...   

                                      review_content  \
0  Looks durable Charging is fine tooNo complains...   
1  I ordered this cable to connect my phone to An...   
2  Not quite durable and sturdy,https://m.media-a...   
3  Good product,long wire,Charges good,Nice,I bou...   
4  Bought this instead of original apple, does th...   

                                            img_link  \
0  https://m.media-amazon.com/images/W/WEBP_40237...   
1  https://m.media-amazon.com/images/W/WEBP_40237...   
2  https://m.media-amazon.com/images/W/WEBP_40237...   
3  https://m.media-amazon.com/images/I/41V5FtEWPk...   
4  https://m.media-amazon.com/images/W/WEBP_40237...   

                                        product_link  
0  https://www.amazon.in/Wayona-Braided-WN3LG1-Sy...  
1  https://www.amazon.in/Ambrane-Unbreakable-Char...  
2  https://www.amazon.in/Sounce-iPhone-Charging-C...  
3  https://www.amazon.in/Deuce-300-Resistant-Tang...  
4  https://www.amazon.in/Portronics-Konnect-POR-1...

## **Let's see the column names**

In [ ]:
df.columns

Index(['product_id', 'product_name', 'category', 'discounted_price',
       'actual_price', 'discount_percentage', 'rating', 'rating_count',
       'about_product', 'user_id', 'user_name', 'review_id', 'review_title',
       'review_content', 'img_link', 'product_link'],
      dtype='object')

## **Let's have a look on the shape of the dataset**

In [ ]:
df.shape

(1465, 16)

## **Let's have a look on the columns and their data types using detailed info function**

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1465 entries, 0 to 1464
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   product_id           1465 non-null   object
 1   product_name         1465 non-null   object
 2   category             1465 non-null   object
 3   discounted_price     1465 non-null   object
 4   actual_price         1465 non-null   object
 5   discount_percentage  1465 non-null   object
 6   rating               1465 non-null   object
 7   rating_count         1463 non-null   object
 8   about_product        1465 non-null   object
 9   user_id              1465 non-null   object
 10  user_name            1465 non-null   object
 11  review_id            1465 non-null   object
 12  review_title         1465 non-null   object
 13  review_content       1465 non-null   object
 14  img_link             1465 non-null   object
 15  product_link         1465 non-null   object
dtypes: obj

In [ ]:
df.isnull().sum()

product_id             0
product_name           0
category               0
discounted_price       0
actual_price           0
discount_percentage    0
rating                 0
rating_count           2
about_product          0
user_id                0
user_name              0
review_id              0
review_title           0
review_content         0
img_link               0
product_link           0
dtype: int64

## **Observation Set 1**
1. There are 1465 rows and 16 columns in the dataset.

2. The data type of all columns is object.

3. The columns in the datasets are:
'product_id', 'product_name', 'category', 'discounted_price', 'actual_price', 'discount_percentage', 'rating', 'rating_count', 'about_product', 'user_id', 'user_name', 'review_id', 'review_title', 'review_content', 'img_link', 'product_link'

4. There are a few missing values in the dataset, which we will read in detail and deal with later on in the notebook.

## **Changing Data Types of Columns from object to float**

In [ ]:
df['discounted_price'] = df['discounted_price'].str.replace('₹', '')
df['discounted_price'] = df['discounted_price'].str.replace(',', '')

In [ ]:
df['discounted_price'] = df['discounted_price'].astype('float')

In [ ]:
df['actual_price'] = df['actual_price'].str.replace('₹', '')
df['actual_price'] = df['actual_price'].str.replace(',', '')
df['actual_price'] = df['actual_price'].astype('float')

In [ ]:
df['discount_percentage'] = df['discount_percentage'].str.replace('%', '')
df['discount_percentage'] = df['discount_percentage'].astype('float')

In [ ]:
df['discount_percentage'] = df['discount_percentage']/100

In [ ]:
df['discount_percentage']

0       0.64
1       0.43
2       0.90
3       0.53
4       0.61
        ... 
1460    0.59
1461    0.25
1462    0.28
1463    0.26
1464    0.22
Name: discount_percentage, Length: 1465, dtype: float64

In [ ]:
df['rating'] = df['rating'].value_counts()

In [ ]:
df['rating'] = copy['rating']

In [ ]:
df['rating'].value_counts()

rating
4.1    244
4.3    230
4.2    228
4.0    129
3.9    123
4.4    123
3.8     86
4.5     75
4       52
3.7     42
3.6     35
3.5     26
4.6     17
3.3     16
3.4     10
4.7      6
3.1      4
3.0      3
4.8      3
5.0      3
2.8      2
3.2      2
2.3      1
|        1
2        1
3        1
2.6      1
2.9      1
Name: count, dtype: int64

In [ ]:
df.query('rating == "|"')

product_id                                       product_name  \
1279  B08L12N5H1  Eureka Forbes car Vac 100 Watts Powerful Sucti...   

                                               category  discounted_price  \
1279  Home&Kitchen|Kitchen&HomeAppliances|Vacuum,Cle...            2099.0   

      actual_price  discount_percentage rating rating_count  \
1279        2499.0                 0.16      |          992   

                                          about_product  \
1279  No Installation is provided for this product|1...   

                                                user_id  \
1279  AGTDSNT2FKVYEPDPXAA673AIS44A,AER2XFSWNN4LAUCJ5...   

                                              user_name  \
1279  Divya,Dr Nefario,Deekshith,Preeti,Prasanth R,P...   

                                              review_id  \
1279  R2KKTKM4M9RDVJ,R1O692MZOBTE79,R2WRSEWL56SOS4,R...   

                                           review_title  \
1279  Decent product,doesn't pick up sand,Ok ok,Must...   

                                         review_content  \
1279  Does the job well,doesn't work on sand. though...   

                                               img_link  \
1279  https://m.media-amazon.com/images/W/WEBP_40237...   

                                           product_link  
1279  https://www.amazon.in/Eureka-Forbes-Vacuum-Cle...

**I got this product rating on Amazon by searching the provided product_id on their official website (amazon.in)**

**The rating is 3.9. So, I am going to give the item rating a 3.9 as well.**

In [ ]:
df['rating'] = df['rating'].str.replace('|','3.9').astype('float')

In [ ]:
df['rating_count'] = df['rating_count'].str.replace(',','').astype('float')

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1465 entries, 0 to 1464
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   product_id           1465 non-null   object 
 1   product_name         1465 non-null   object 
 2   category             1465 non-null   object 
 3   discounted_price     1465 non-null   float64
 4   actual_price         1465 non-null   float64
 5   discount_percentage  1465 non-null   float64
 6   rating               1465 non-null   float64
 7   rating_count         1463 non-null   float64
 8   about_product        1465 non-null   object 
 9   user_id              1465 non-null   object 
 10  user_name            1465 non-null   object 
 11  review_id            1465 non-null   object 
 12  review_title         1465 non-null   object 
 13  review_content       1465 non-null   object 
 14  img_link             1465 non-null   object 
 15  product_link         1465 non-null   o

## **Descriptive Statistics**

Descriptive statistics are a collection of quantitative measures that summarize and describe the main characteristics of a dataset.

In [ ]:
df.describe()

discounted_price   actual_price  discount_percentage       rating  \
count       1465.000000    1465.000000          1465.000000  1465.000000   
mean        3125.310874    5444.990635             0.476915     4.096451   
std         6944.304394   10874.826864             0.216359     0.291620   
min           39.000000      39.000000             0.000000     2.000000   
25%          325.000000     800.000000             0.320000     4.000000   
50%          799.000000    1650.000000             0.500000     4.100000   
75%         1999.000000    4295.000000             0.630000     4.300000   
max        77990.000000  139900.000000             0.940000     5.000000   

        rating_count  
count    1463.000000  
mean    18295.541353  
std     42753.864952  
min         2.000000  
25%      1186.000000  
50%      5179.000000  
75%     17336.500000  
max    426973.000000

## **Observation Set 2**

1. All columns data type was object So, I converted some column data type to float.

2. There are 4 numeric as per Python coding or descriptive statistics from Python describe function

## **Dealing with the missing values**

Dealing with the missing values is one of the most important part of the data wrangling process, we must deal with the missing values in order to get the correct insights from the data.¶


## **Missing Values**

In [ ]:
df.isnull().sum()

product_id             0
product_name           0
category               0
discounted_price       0
actual_price           0
discount_percentage    0
rating                 0
rating_count           2
about_product          0
user_id                0
user_name              0
review_id              0
review_title           0
review_content         0
img_link               0
product_link           0
dtype: int64

In [ ]:
df[df['rating_count'].isnull()]

product_id                                       product_name  \
282  B0B94JPY2N  Amazon Brand - Solimo 65W Fast Charging Braide...   
324  B0BQRJ3C47  REDTECH USB-C to Lightning Cable 3.3FT, [Apple...   

                                              category  discounted_price  \
282  Computers&Accessories|Accessories&Peripherals|...             199.0   
324  Computers&Accessories|Accessories&Peripherals|...             249.0   

     actual_price  discount_percentage  rating  rating_count  \
282         999.0                 0.80     3.0           NaN   
324         999.0                 0.75     5.0           NaN   

                                         about_product  \
282  USB C to C Cable: This cable has type C connec...   
324  💎[The Fastest Charge] - This iPhone USB C cabl...   

                          user_id    user_name      review_id  \
282  AE7CFHY23VAJT2FI4NZKKP6GS2UQ       Pranav   RUB7U91HVZ30   
324  AGJC5O5H5BBXWUV7WRIEIOOR3TVQ  Abdul Gafur  RQXD5SAMMPC6L   

                                     review_title  \
282  The cable works but is not 65W as advertised   
324                               Awesome Product   

                                        review_content  \
282  I have a pd supported car charger and I bought...   
324  Quick delivery.Awesome ProductPacking was good...   

                                              img_link  \
282  https://m.media-amazon.com/images/W/WEBP_40237...   
324  https://m.media-amazon.com/images/I/31-q0xhaTA...   

                                          product_link  
282  https://www.amazon.in/Amazon-Brand-Charging-Su...  
324  https://www.amazon.in/REDTECH-Lightning-Certif...

In [ ]:
df['rating_count'] = df['rating'].fillna(df['rating_count'].median())

In [ ]:
df.isnull().sum()

product_id             0
product_name           0
category               0
discounted_price       0
actual_price           0
discount_percentage    0
rating                 0
rating_count           0
about_product          0
user_id                0
user_name              0
review_id              0
review_title           0
review_content         0
img_link               0
product_link           0
dtype: int64

In [ ]:
df.duplicated().any()

np.False_

## **Simplifying Category Column**

In [ ]:
# First, let's see the structure of your category data
print("Sample categories:")
print(df['category'].head(10))
print("\nMax number of levels in categories:")
max_levels = df['category'].str.count('\|').max() + 1
print(f"Maximum levels found: {max_levels}")

Sample categories:
0    Computers&Accessories|Accessories&Peripherals|...
1    Computers&Accessories|Accessories&Peripherals|...
2    Computers&Accessories|Accessories&Peripherals|...
3    Computers&Accessories|Accessories&Peripherals|...
4    Computers&Accessories|Accessories&Peripherals|...
5    Computers&Accessories|Accessories&Peripherals|...
6    Computers&Accessories|Accessories&Peripherals|...
7    Computers&Accessories|Accessories&Peripherals|...
8    Computers&Accessories|NetworkingDevices|Networ...
9    Computers&Accessories|Accessories&Peripherals|...
Name: category, dtype: object

Max number of levels in categories:
Maximum levels found: 7


<>:5: SyntaxWarning: invalid escape sequence '\|'
<>:5: SyntaxWarning: invalid escape sequence '\|'
/tmp/ipython-input-4009054515.py:5: SyntaxWarning: invalid escape sequence '\|'
  max_levels = df['category'].str.count('\|').max() + 1


In [ ]:
# Simple split - just take the first 2 levels
df[['main_category', 'Subcategory']] = df['category'].str.split('|', n=1, expand=True)



# Check results
print(df['main_category'].value_counts())

main_category
Electronics              526
Computers&Accessories    453
Home&Kitchen             448
OfficeProducts            31
MusicalInstruments         2
HomeImprovement            2
Toys&Games                 1
Car&Motorbike              1
Health&PersonalCare        1
Name: count, dtype: int64


In [ ]:
df.head(3)

product_id                                       product_name  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   

                                            category  discounted_price  \
0  Computers&Accessories|Accessories&Peripherals|...             399.0   
1  Computers&Accessories|Accessories&Peripherals|...             199.0   
2  Computers&Accessories|Accessories&Peripherals|...             199.0   

   actual_price  discount_percentage  rating  rating_count  \
0        1099.0                 0.64     4.2           4.2   
1         349.0                 0.43     4.0           4.0   
2        1899.0                 0.90     3.9           3.9   

                                       about_product  \
0  High Compatibility : Compatible With iPhone 12...   
1  Compatible with all Type C enabled devices, be...   
2  【 Fast Charger& Data Sync】-With built-in safet...   

                                             user_id  \
0  AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1  AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2  AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   

                                           user_name  \
0  Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...   
1  ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...   
2  Kunal,Himanshu,viswanath,sai niharka,saqib mal...   

                                           review_id  \
0  R3HXWT0LRP0NMF,R2AJM3LFTLZHFO,R6AQJGUP6P86,R1K...   
1  RGIQEG07R9HS2,R1SMWZQ86XIN8U,R2J3Y1WL29GWDE,RY...   
2  R3J3EQQ9TZI5ZJ,R3E7WBGK7ID0KV,RWU79XKQ6I1QF,R2...   

                                        review_title  \
0  Satisfied,Charging is really fast,Value for mo...   
1  A Good Braided Cable for Your Type C Device,Go...   
2  Good speed for earlier versions,Good Product,W...   

                                      review_content  \
0  Looks durable Charging is fine tooNo complains...   
1  I ordered this cable to connect my phone to An...   
2  Not quite durable and sturdy,https://m.media-a...   

                                            img_link  \
0  https://m.media-amazon.com/images/W/WEBP_40237...   
1  https://m.media-amazon.com/images/W/WEBP_40237...   
2  https://m.media-amazon.com/images/W/WEBP_40237...   

                                        product_link          main_category  \
0  https://www.amazon.in/Wayona-Braided-WN3LG1-Sy...  Computers&Accessories   
1  https://www.amazon.in/Ambrane-Unbreakable-Char...  Computers&Accessories   
2  https://www.amazon.in/Sounce-iPhone-Charging-C...  Computers&Accessories   

                                         Subcategory  
0  Accessories&Peripherals|Cables&Accessories|Cab...  
1  Accessories&Peripherals|Cables&Accessories|Cab...  
2  Accessories&Peripherals|Cables&Accessories|Cab...

## **Simplifying Product Column**

In [ ]:
# First, let's see what product types actually exist in your data
# This will help us identify all the patterns we need to cover

# Check most common words in product names
from collections import Counter
import re

# Extract all words from product names
all_words = []
for name in df['product_name'].dropna():
    words = re.findall(r'\b\w+\b', name.lower())
    all_words.extend(words)

# Count most frequent words
word_counts = Counter(all_words)
print("Top 20 most common words in product names:")
for word, count in word_counts.most_common(50):
    print(f"{word}: {count}")

Top 20 most common words in product names:
with: 752
for: 674
black: 530
usb: 417
cable: 414
1: 413
and: 331
c: 289
3: 275
2: 267
charging: 247
type: 243
to: 219
fast: 212
in: 178
smart: 175
white: 170
tv: 152
compatible: 146
5: 137
data: 136
hd: 135
4: 133
led: 129
8: 125
pro: 125
0: 124
wireless: 119
water: 114
10: 113
6: 109
remote: 109
heater: 108
bluetooth: 107
grey: 106
laptop: 105
of: 101
battery: 97
ram: 96
electric: 96
braided: 94
sync: 92
iphone: 92
adapter: 91
blue: 90
watch: 89
watt: 89
ultra: 87
storage: 86
samsung: 85


In [ ]:
# Advanced product categorization based on your actual top words
def get_product_type_advanced(name):
    name_lower = name.lower()

    # Cables & Connectivity (cable, usb, charging, type, data, sync, adapter)
    if any(word in name_lower for word in ['cable', 'usb', 'charging', 'adapter', 'sync', 'data']):
        return 'Cables & Connectivity'

    # Smart TV & Electronics (tv, smart, led, hd, remote, wireless)
    elif any(word in name_lower for word in ['tv', 'smart', 'led', 'hd', 'remote']):
        return 'TV & Smart Devices'

    # Mobile & Computing (iphone, samsung, laptop, ram, storage, pro, bluetooth)
    elif any(word in name_lower for word in ['iphone', 'samsung', 'laptop', 'ram', 'storage', 'bluetooth']):
        return 'Mobile & Computing'

    # Home Appliances (water, heater, electric, watt)
    elif any(word in name_lower for word in ['water', 'heater', 'electric', 'watt']):
        return 'Home Appliances'

    # Wearables & Accessories (watch, battery, wireless)
    elif any(word in name_lower for word in ['watch', 'battery', 'wireless']):
        return 'Wearables & Accessories'

    else:
        return 'Other Electronics'

# Apply the classification
df['product_category_new'] = df['product_name'].apply(get_product_type_advanced)

# Check results
print("Advanced product categorization:")
print(df['product_category_new'].value_counts())

# Check coverage
other_percentage = (df['product_category_new'] == 'Other Electronics').sum() / len(df) * 100
print(f"\nItems in 'Other Electronics': {other_percentage:.1f}%")

Advanced product categorization:
product_category_new
Cables & Connectivity      477
TV & Smart Devices         324
Other Electronics          246
Home Appliances            234
Mobile & Computing         156
Wearables & Accessories     28
Name: count, dtype: int64

Items in 'Other Electronics': 16.8%


In [ ]:
df.head(3)

product_id                                       product_name  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   

                                            category  discounted_price  \
0  Computers&Accessories|Accessories&Peripherals|...             399.0   
1  Computers&Accessories|Accessories&Peripherals|...             199.0   
2  Computers&Accessories|Accessories&Peripherals|...             199.0   

   actual_price  discount_percentage  rating  rating_count  \
0        1099.0                 0.64     4.2           4.2   
1         349.0                 0.43     4.0           4.0   
2        1899.0                 0.90     3.9           3.9   

                                       about_product  \
0  High Compatibility : Compatible With iPhone 12...   
1  Compatible with all Type C enabled devices, be...   
2  【 Fast Charger& Data Sync】-With built-in safet...   

                                             user_id  \
0  AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1  AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2  AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   

                                           user_name  \
0  Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...   
1  ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...   
2  Kunal,Himanshu,viswanath,sai niharka,saqib mal...   

                                           review_id  \
0  R3HXWT0LRP0NMF,R2AJM3LFTLZHFO,R6AQJGUP6P86,R1K...   
1  RGIQEG07R9HS2,R1SMWZQ86XIN8U,R2J3Y1WL29GWDE,RY...   
2  R3J3EQQ9TZI5ZJ,R3E7WBGK7ID0KV,RWU79XKQ6I1QF,R2...   

                                        review_title  \
0  Satisfied,Charging is really fast,Value for mo...   
1  A Good Braided Cable for Your Type C Device,Go...   
2  Good speed for earlier versions,Good Product,W...   

                                      review_content  \
0  Looks durable Charging is fine tooNo complains...   
1  I ordered this cable to connect my phone to An...   
2  Not quite durable and sturdy,https://m.media-a...   

                                            img_link  \
0  https://m.media-amazon.com/images/W/WEBP_40237...   
1  https://m.media-amazon.com/images/W/WEBP_40237...   
2  https://m.media-amazon.com/images/W/WEBP_40237...   

                                        product_link          main_category  \
0  https://www.amazon.in/Wayona-Braided-WN3LG1-Sy...  Computers&Accessories   
1  https://www.amazon.in/Ambrane-Unbreakable-Char...  Computers&Accessories   
2  https://www.amazon.in/Sounce-iPhone-Charging-C...  Computers&Accessories   

                                         Subcategory   product_category_new  
0  Accessories&Peripherals|Cables&Accessories|Cab...  Cables & Connectivity  
1  Accessories&Peripherals|Cables&Accessories|Cab...  Cables & Connectivity  
2  Accessories&Peripherals|Cables&Accessories|Cab...  Cables & Connectivity

In [ ]:
# Even simpler approach - just truncate and clean basic stuff
def simple_clean(name):
    if pd.isna(name):
        return name

    # Remove common noise
    name = name.replace('Amazon Basics', '').replace('(', '').replace(')', '')
    name = name.replace('[', '').replace(']', '')

    # Take first part before excessive details
    if 'Compatible with' in name:
        name = name.split('Compatible with')[0]
    if ' - ' in name:
        name = name.split(' - ')[0]

    # Truncate and clean
    return name[:45].strip()

df['product_name_simple'] = df['product_name'].apply(simple_clean)

print("Simplified names:")
print(df[['product_name', 'product_name_simple']].head())

Simplified names:
                                        product_name  \
0  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  Sounce Fast Phone Charging Cable & Data Sync U...   
3  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   

                             product_name_simple  
0  Wayona Nylon Braided USB to Lightning Fast Ch  
1  Ambrane Unbreakable 60W / 3A Fast Charging 1.  
2   Sounce Fast Phone Charging Cable & Data Sync  
3   boAt Deuce USB 300 2 in 1 Type-C & Micro USB  
4   Portronics Konnect L 1.2M Fast Charging 3A 8  


In [ ]:
df.columns

Index(['product_id', 'product_name', 'category', 'discounted_price',
       'actual_price', 'discount_percentage', 'rating', 'rating_count',
       'about_product', 'user_id', 'user_name', 'review_id', 'review_title',
       'review_content', 'img_link', 'product_link', 'main_category',
       'Subcategory', 'product_category_new', 'product_name_simple'],
      dtype='object')

In [ ]:
df.drop(['product_category_new', 'product_name_simple'],axis=1)

product_id                                       product_name  \
0     B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1     B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2     B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3     B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4     B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   
...          ...                                                ...   
1460  B08L7J3T31  Noir Aqua - 5pcs PP Spun Filter + 1 Spanner | ...   
1461  B01M6453MB  Prestige Delight PRWO Electric Rice Cooker (1 ...   
1462  B009P2LIL4  Bajaj Majesty RX10 2000 Watts Heat Convector R...   
1463  B00J5DYCCA  Havells Ventil Air DSP 230mm Exhaust Fan (Pist...   
1464  B01486F4G6  Borosil Jumbo 1000-Watt Grill Sandwich Maker (...   

                                               category  discounted_price  \
0     Computers&Accessories|Accessories&Peripherals|...             399.0   
1     Computers&Accessories|Accessories&Peripherals|...             199.0   
2     Computers&Accessories|Accessories&Peripherals|...             199.0   
3     Computers&Accessories|Accessories&Peripherals|...             329.0   
4     Computers&Accessories|Accessories&Peripherals|...             154.0   
...                                                 ...               ...   
1460  Home&Kitchen|Kitchen&HomeAppliances|WaterPurif...             379.0   
1461  Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...            2280.0   
1462  Home&Kitchen|Heating,Cooling&AirQuality|RoomHe...            2219.0   
1463  Home&Kitchen|Heating,Cooling&AirQuality|Fans|E...            1399.0   
1464  Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...            2863.0   

      actual_price  discount_percentage  rating  rating_count  \
0           1099.0                 0.64     4.2           4.2   
1            349.0                 0.43     4.0           4.0   
2           1899.0                 0.90     3.9           3.9   
3            699.0                 0.53     4.2           4.2   
4            399.0                 0.61     4.2           4.2   
...            ...                  ...     ...           ...   
1460         919.0                 0.59     4.0           4.0   
1461        3045.0                 0.25     4.1           4.1   
1462        3080.0                 0.28     3.6           3.6   
1463        1890.0                 0.26     4.0           4.0   
1464        3690.0                 0.22     4.3           4.3   

                                          about_product  \
0     High Compatibility : Compatible With iPhone 12...   
1     Compatible with all Type C enabled devices, be...   
2     【 Fast Charger& Data Sync】-With built-in safet...   
3     The boAt Deuce USB 300 2 in 1 cable is compati...   
4     [CHARGE & SYNC FUNCTION]- This cable comes wit...   
...                                                 ...   
1460  SUPREME QUALITY 90 GRAM 3 LAYER THIK PP SPUN F...   
1461                       230 Volts, 400 watts, 1 Year   
1462  International design and styling|Two heat sett...   
1463  Fan sweep area: 230 MM ; Noise level: (40 - 45...   
1464  Brand-Borosil, Specification – 23V ~ 5Hz;1 W...   

                                                user_id  \
0     AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1     AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2     AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3     AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4     AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   
...                                                 ...   
1460  AHITFY6AHALOFOHOZEOC6XBP4FEA,AFRABBODZJZQB6Z4U...   
1461  AFG5FM3NEMOL6BNFRV2NK5FNJCHQ,AGEINTRN6Z563RMLH...   
1462  AGVPWCMAHYQWJOQKMUJN4DW3KM5Q,AF4Q3E66MY4SR7YQZ...   
1463  AF2JQCLSCY3QJATWUNNHUSVUPNQQ,AFDMLUXC5LS5RXDJS...   
1464  AFGW5PT3R6ZAVQR4Y5MWVAKBZAYA,AG7QNJ2SCS5VS5VYY...   

                             

## **Adding Performance Column based on Rating**

In [ ]:
# Create performance categories based on rating
def categorize_performance(rating):
    if pd.isna(rating):
        return 'Unknown'
    elif rating >= 4.0:
        return 'Excellent'
    elif rating >= 3.0:
        return 'Average'
    else:
        return 'Bad'

df['performance'] = df['rating'].apply(categorize_performance)

# Check the distribution
print("Performance distribution:")
print(df['performance'].value_counts())

# Check percentage breakdown
print("\nPercentage breakdown:")
print(df['performance'].value_counts(normalize=True) * 100)

Performance distribution:
performance
Excellent    1110
Average       349
Bad             6
Name: count, dtype: int64

Percentage breakdown:
performance
Excellent    75.767918
Average      23.822526
Bad           0.409556
Name: proportion, dtype: float64


In [ ]:
df = df.drop(['product_category_new', 'product_name_simple'],axis=1)

In [ ]:
df.columns

Index(['product_id', 'product_name', 'category', 'discounted_price',
       'actual_price', 'discount_percentage', 'rating', 'rating_count',
       'about_product', 'user_id', 'user_name', 'review_id', 'review_title',
       'review_content', 'img_link', 'product_link', 'main_category',
       'Subcategory', 'performance'],
      dtype='object')

In [ ]:
df.to_excel('AmazonSales1.xlsx')